main.m

```diff
@@ -170,29 +177,39 @@
 	  // skip existing GNUmakefiles
 	  if ([projectDirEntry hasPrefix: @"GNUmakefile"])
 	    continue;
-					       
+    
+    // If not generating the pbxbuild directory...
+    if (args_info.generate_flat_directory_flag == 1)
```

Comment:
Is there any possibility for this flag to not be a zero or one? If so, I think the comparison is a bit redundant for a boolean variable.

Reply:
This is where software coding standards apply.  Nothing should *ever* be assumed, and even booleans should be checked literally.  

The one that I should do is reverse the check - it should be:
1 == varName
as opposed to the current coding - however I applied the current coding since that is what the rest of the code uses.

Believe it or not - your suggestion would not pass a code review since it would violate typical standards used by hard core companies.

Reply:
Than line 116 needs changed to check if == 1 in this file. And thanks for that knowledge dump! That is good to know the standard I was using does not pass snuff.